diff --git a/wxgraphs.php b/wxgraphs.php
index 90531c0..b57fb12 100644
--- a/wxgraphs.php
+++ b/wxgraphs.php
@@ -381,6 +381,142 @@ print "</p>\n";
     
 <?php } // end wview graph names    ?>
 
+<!--
+//-------------------------------------------------------------------------
+// weewx using the weewx-wd extension to generate WD-compatible input files
+//--------------------------------------------------------------------------
+-->
+
+<?php if($SITE['WXsoftware'] == 'WE') { // use weewx names of images and noaa files
+   $WVgraphDescs = array(
+      'd' => 'Daily',
+ 	 'w' => 'Weekly',
+ 	 'm' => 'Monthly',
+ 	 'y' => 'Yearly',
+   );
+   $WVgraphPeriods = array(
+      'd' => '24h',
+ 	 'w' => '7d',
+ 	 'm' => '28d',
+ 	 'y' => '365d',
+   );
+ 
+ $WVgraphs = array(
+ # 24h graphs
+ '24h-temp' => 'daytempdew.png',
+ '24h-intemp' => 'dayinside.png',
+ '24h-rx' => 'dayrx.png',
+ '24h-hc' => 'daytempchill.png',
+ #'24h-hum' => 'daytempdew.png',
+ '24h-baro' => 'daybarometer.png',
+ '24h-rain' => 'dayrain.png',
+ '24h-wdir' => 'daywinddir.png',
+ '24h-wsp' => 'daywind.png',
+ #'24h-rad' => 'radiationDay.png',
+ #'24h-uv' => 'UVDay.png',
+ 
+ # 7d graphs
+ '7d-temp' => 'weektempdew.png',
+ '7d-intemp' => 'weekinside.png',
+ '7d-rx' => 'weekrx.png',
+ '7d-hc' => 'weektempchill.png',
+ #'7d-hum' => 'weektempdew.png',
+ '7d-baro' => 'weekbarometer.png',
+ '7d-rain' => 'weekrain.png',
+ '7d-wdir' => 'weekwinddir.png',
+ '7d-wsp' => 'weekwind.png',
+ #'7d-rad' => 'radiationDay.png',
+ #'7d-uv' => 'UVDay.png',
+ 
+ # 28d graphs
+ '28d-temp' => 'monthtempdew.png',
+ '28d-intemp' => 'monthinside.png',
+ '28d-rx' => 'monthrx.png',
+ '28d-hc' => 'monthtempchill.png',
+ #'28d-hum' => 'monthtempdew.png',
+ '28d-baro' => 'monthbarometer.png',
+ '28d-rain' => 'monthrain.png',
+ '28d-wdir' => 'monthwinddir.png',
+ '28d-wsp' => 'monthwind.png',
+ #'28d-rad' => 'radiationDay.png',
+ #'28d-uv' => 'UVDay.png',
+ 
+ # 365d graphs
+ '365d-temp' => 'yeartempdew.png',
+ '365d-intemp' => 'yearinside.png',
+ '365d-rx' => 'yearrx.png',
+ '365d-hc' => 'yeartempchill.png',
+ #'365d-hum' => 'yeartempdew.png',
+ '365d-baro' => 'yearbarometer.png',
+ '365d-rain' => 'yearrain.png',
+ '365d-wdir' => 'yearwinddir.png',
+ '365d-wsp' => 'yearwind.png',
+ #'365d-rad' => 'radiationDay.png',
+ #'365d-uv' => 'UVDay.png',
+ 
+ );
+ 
+ //$WVgraphIdx = 'w';  // default is 7-day set
+ $WVgraphIdx = 'd';  // default is one-day set
+ $WVgraphName = $WVgraphDescs[$WVgraphIdx];
+ $WVgraphSet =  $WVgraphPeriods[$WVgraphIdx];
+ 
+ 
+ if(isset($_REQUEST['graph'])) { // check out selection
+   $req = strtolower($_REQUEST['graph']);
+   foreach ($WVgraphDescs as $i => $v) {
+ 	  if ($req == $i) {
+ 		$WVgraphIdx = $i;  // default is 7-day set
+ 		$WVgraphName = $WVgraphDescs[$i];
+ 		$WVgraphSet =  $WVgraphPeriods[$i];
+ 		break;  
+ 	  }
+   }
+ }
+ // generate the links for the available periods to display
+ print "<p style=\"text-align: center\">";
+ foreach ($WVgraphDescs as $i => $v) {
+ 	if ($i == $WVgraphIdx) {
+ 		print "<strong>".langtransstr($v)."</strong>&nbsp;&nbsp;";
+ 	} else {
+ 		print "<a href=\"?graph=$i\">".langtransstr($v)."</a>&nbsp;&nbsp;";
+ 	}
+ }
+ print "</p>\n";	
+  ?>
+     <h2><?php echo langtransstr('Temperature'); ?></h2>
+ 	<?php genImageLink($WVgraphs[$WVgraphSet."-temp"],'Outside Temperature'); ?>
+ 	<?php genImageLink($WVgraphs[$WVgraphSet."-intemp"],'Inside Temperature'); ?>
+     <br />
+     <h2><?php echo langtransstr('Rain') . ' / ' . langtransstr('Barometer'); ?></h2>
+     <?php genImageLink($WVgraphs[$WVgraphSet."-rain"],'Rain'); ?>
+     <?php genImageLink($WVgraphs[$WVgraphSet."-baro"],'Barometer'); ?>
+     <br />
+     <h2><?php echo langtransstr('Wind'); ?></h2>
+     <?php genImageLink($WVgraphs[$WVgraphSet."-wsp"],'Wind Speed'); ?>
+     <?php genImageLink($WVgraphs[$WVgraphSet."-wdir"],'Wind Direction'); ?>
+     <br />
+     <h2><?php langtrans('Wind Chill'); ?></h2>
+     <?php genImageLink($WVgraphs[$WVgraphSet."-hc"],'Wind Chill'); ?>
+     <br />
+     <h2><?php langtrans('ISS Signal Quality'); ?></h2>
+     <?php genImageLink($WVgraphs[$WVgraphSet."-rx"],'Signal Quality'); ?>
+     <br />
+     <?php // figure out to display solar and/or UV based on site settings
+ 	if($SITE['SOLAR'] and $SITE['UV']) { // have both sensors ?>
+     <h2><?php echo langtransstr('Solar Radiation') . ' / ' . langtransstr('UV Index'); ?></h2>
+     <?php genImageLink($WVgraphs[$WVgraphSet."-rad"],'Solar Radiation'); ?>
+     <?php genImageLink($WVgraphs[$WVgraphSet."-uv"],'UV Index'); ?>
+     <?php } elseif ($SITE['SOLAR']) { ?>
+      <h2><?php echo langtransstr('Solar Radiation'); ?></h2>
+     <?php genImageLink($WVgraphs[$WVgraphSet."-rad"],'Solar Radiation'); ?>
+     <?php } elseif ($SITE['UV']) { ?>
+      <h2><?php echo langtransstr('UV Index'); ?></h2>
+     <?php genImageLink($WVgraphs[$WVgraphSet."-uv"],'UV Index'); ?>
+     <?php } // end solar/uv selection ?>
+ 
+     
+ <?php } // end weewx-wd graph names    ?>
 
 </div><!-- end main-copy -->
 
@@ -408,4 +544,4 @@ include("footer.php");
 ############################################################################
 # End of Page
 ############################################################################
-?>
\ No newline at end of file
+?>
